viadot/sources/vid_club.py

```diff
@@ -0,0 +1,165 @@
+import json
+import urllib
+import os
+
+from typing import Any, Dict, List, Literal
+from datetime import datetime
+
+import pandas as pd
+
+from ..config import local_config
+from ..exceptions import CredentialError, ValidationError
+from ..utils import handle_api_response
+from .base import Source
+
+
+class VidClub(Source):
+    """
+    A class implementing the Vid Club API.
+
+    Documentation for this API is located at: https://evps01.envoo.net/vipapi/
+    There are 4 endpoints where to get the data.
+
+    """
+
+    def __init__(self, *args, credentials: Dict[str, Any] = None, **kwargs):
+        """
+        Create an instance of VidClub.
+
+        Args:
+            credentials (dict, optional): Credentials to Vid Club APIs.
+                Defaults to dictionary.
+        """
+
+        DEFAULT_CREDENTIALS = local_config.get("VIDCLUB")
+        credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
+        if credentials is None:
+            raise CredentialError("Missing credentials.")
+
+        self.headers = {
+            "Authorization": "Bearer " + credentials["token"],
+            "Content-Type": "application/json",
+        }
+
+        super().__init__(*args, credentials=credentials, **kwargs)
+
+    def build_query(
+        self,
+        source: Literal["jobs", "product", "company", "survey"],
+        from_date: str,
+        to_date: str,
+        api_url: str,
+        items_per_page: int,
+    ) -> str:
+        """
+        Builds the query from the inputs.
+
+        Args:
+            source (str): The endpoint source to be accessed, has to be among these:
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str): Start date for the query.
+            to_date (str): End date for the query, if empty, datetime.today() will be used.
+            api_url (str): Generic part of the URL.
+            items_per_page (int): number of entries per page.
+
+        Returns:
+            str: Final query with all filters added.
```

Comment:
Please add `Raises` section in the docstring. In this function, you have `ValidationError`.